### ISSUE TITLE
Hatnote is erroneously moved to second paragraph along with infobox 

### ISSUE THREAD

**Issue by Jaifroid** [openzim/mwoffliner#182]
Viewing the article "Peripheral neuropathy" in `wikipedia_en_medicine_novid_2018-01.zim`, the hatnote "Not to be confused with..." has been shifted to the second paragraph along with the infobox (see first screenshot). On Wikipedia mobile view this line is in the correct place just below the page title. I'm guessing this is a mwoffliner issue, as it's in the HTML that comes out of the ZIM. 

It's a general issue, because it also occurs with the "Melatonin" article (see second screenshot).

![image](https://user-images.githubusercontent.com/4304337/34832299-6f2f9f32-f6e1-11e7-8c55-1b0941db5a66.png)

![image](https://user-images.githubusercontent.com/4304337/34832579-64ff0a10-f6e2-11e7-8731-cc45b5ba1da0.png)

**Comment by kelson42** [comment]
@subbuss If I look in Parsoid output, the hatnot is put in the middle (like in the ZIM). But this is not how the online version behaves, like @Jaifroid reported. Looks strange for me. Do we have a bug in Parsoid here? Or do we need to make special handling?

**Comment by subbuss** [comment]
https://en.wikipedia.org/api/rest_v1/page/html/Peripheral_neuropathy shows it at the right place?

**Comment by kelson42** [comment]
@subbuss for the desktop version yes, but not for the mobile. See https://en.wikipedia.org/api/rest_v1/page/mobile-sections/Peripheral_neuropathy

**Comment by mdholloway** [comment]
For the currently deployed Wikimedia REST API mobile-sections endpoint, moving the initial paragraph of an article up top, even above hatnotes, is actually the intended behavior.  This was driven by a design decision for the Wikipedia Android app, which was the endpoint's initial consumer.

In the next-generation (not-yet-active) version of mobile-sections the hatnote is broken out into the response json structure greater client flexibility.

**Comment by kelson42** [comment]
@mdholloway thx. So basically in the future version the hatnote will be removed from article HTML and will be only available as metadata (like it is already available currently)?

**Comment by Jaifroid** [comment]
Just to add that the same issue affects notes (in English Wikivoyage) of the type "For other places with the same name, see Paris (disambiguation)" (this is from the Paris article). However, this note isn't labelled "hatnote", it is inside a `<dl>` structure and it is identified with a CSS class of "noexcerpt". This is what the HTML looks like:
```
<dl>
  <dd>
    <span class="noexcerpt"><i>For other places with the same name, see 
      <a href="Paris_(disambiguation).html" title="Paris (disambiguation)">
      Paris (disambiguation)</a>.</i>
    </span>
  </dd>
</dl>
```
And this is the screenshot:

![image](https://user-images.githubusercontent.com/4304337/34908754-b89068a8-f88c-11e7-9001-ec940add6062.png)

**Comment by Jaifroid** [comment]
Just a quick note to say this affects also French Wikipedia / WikiMed which has some "hatnote" equivalents which are rendered as `<div class="homonymie" ...>`. Screenshot shows one of these. It seems that whatever code moves the infoboxes down below the lead paragraph (in order to produce the mobile style) is accidentally but systematically moving hatnotes along with the infobox. Probably a regex that is not specific enough.

![image](https://user-images.githubusercontent.com/4304337/36775756-3d9c7338-1c5b-11e8-9119-3973c12b7943.png)

**Comment by bradyhunsaker** [comment]
@kelson42 I think this one gets the Parsoid tag, since my reading is that there are no plans to change mwoffliner based on the current Parsoid output.  (It looks like that would be an awkward hack to even try.)

**Comment by Jaifroid** [comment]
Just a quick update to note that I'm still getting this error on recent ZIMs. Example below is from `wikipedia_en_maths_novid_2018-06.zim`, article "Series (mathematics)". I realize we may well be waiting on a change in Parsoid, but it would be good to prevent this issue from being put on the back burner...

![image](https://user-images.githubusercontent.com/4304337/45650113-2c56a680-bac5-11e8-9453-40f4e400064c.png)

**Comment by kelson42** [comment]
Looks like hatnotes are completly removed now... Not sure if this is good or bad, but now this problem does not occur anymore.

**Comment by Jaifroid** [comment]
It may not be labelled "hatnote", but in `wikivoyage_en_all_novid_2019-07.zim` (made very recently), we still have a similar problem:

![image](https://user-images.githubusercontent.com/4304337/61629234-36fd4180-ac7c-11e9-987d-90cc8a69304f.png)

Do you want me to make a new issue for it, @kelson42, or re-open this one?

**Comment by Jaifroid** [comment]
Just to say that this issue persists in `wikipedia_en_all_maxi_2020-06.zim`:

![image](https://user-images.githubusercontent.com/4304337/87885112-345a3700-ca0b-11ea-88cb-bcc062bca347.png)

**Comment by Jaifroid** [comment]
Just to keep this alive, issue still persists. In Kiwix JS Windows / PWA, I currently reposition these misplaced hatnotes, redirects and other "not to be confused with" notes, though it's a little tricky because they're not always easy to identify programmatically, and I only test in English, Spanish, German, occasionally French...

![image](https://user-images.githubusercontent.com/4304337/124482374-1e22c380-dda1-11eb-8904-63a1a3865d21.png)

**Comment by kelson42** [comment]
Bug is still there http://library.kiwix.org/wikipedia_en_medicine/A/Peripheral_neuropathy

**Comment by Jaifroid** [comment]
Issue persists (also, the infobox is poorly rendered on desktop screen sizes. It should be right-aligned and thinner on large screens; on narrow screens it is not rendered on Wikipedia mobile view).

I try to fix these display issues in Kiwix JS Windows (see screenshot far bottom). I know it's hacky, and not really the reader's job, but it seems this issue can't be fixed in the ZIM, and won't be fixed by Parsoid either...

![image](https://user-images.githubusercontent.com/4304337/133944968-e26a3020-ae5b-403a-880f-dddddd828762.png)

![2021-09-19 (3)](https://user-images.githubusercontent.com/4304337/133945145-3a723401-c4a3-491e-8bf3-220cd9e68a6f.png)

**Comment by kelson42** [comment]
This has been fixed (with ActionParse), see https://browse.library.kiwix.org/viewer#wikipedia_en_medicine_maxi_2025-03/A/Peripheral_neuropathy
